From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Paramjot (PJ) Oberoi" <60081609+pjsoberoi@users.noreply.github.com>
Date: Fri, 22 Jul 2022 21:44:41 -0400
Subject: [PATCH] 4481: V850: Added Extended Floating Point Instructions

V850: fmaf.s, fnmfs.s Instructions

Added support for the fmaf.s and fnmfs.s instructions.

V850: Added Extended Floating Point Instructions

Added fmaf.s, fmsf.s, fnmaf.s fnmfs.s instructions

Remove Extra Newline

No code change
---
 .../data/languages/Instructions/Float.sinc    | 24 +++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/Ghidra/Processors/V850/data/languages/Instructions/Float.sinc b/Ghidra/Processors/V850/data/languages/Instructions/Float.sinc
index 60f3b2bc1..fef4edb82 100644
--- a/Ghidra/Processors/V850/data/languages/Instructions/Float.sinc
+++ b/Ghidra/Processors/V850/data/languages/Instructions/Float.sinc
@@ -294,6 +294,30 @@
 	R2731 = trunc(floor(R1115));
 }
 
+# FMAF.S reg1, reg2, reg3 - rrrrr111111RRRRR|wwwww10011100000
+:fmaf.s R0004, R1115, R2731 is R0004 & op0510=0x3F & R1115; R2731 & op1626=0b10011100000
+{
+	R2731 = (R1115 f* R0004) f+ R2731;
+}
+
+# FMSF.S reg1, reg2, reg3 - rrrrr111111RRRRR|wwwww10011100010
+:fmsf.s R0004, R1115, R2731 is R0004 & op0510=0x3F & R1115; R2731 & op1626=0b10011100010
+{
+	R2731 = (R1115 f* R0004) f- R2731;
+}
+
+# FNMAF.S reg1, reg2, reg3 - rrrrr111111RRRRR|wwwww10011100100
+:fnmaf.s R0004, R1115, R2731 is R0004 & op0510=0x3F & R1115; R2731 & op1626=0b10011100100
+{
+	R2731 = -1 f* ((R1115 f* R0004) f+ R2731);
+}
+
+# FNMSF.S reg1, reg2, reg3 - rrrrr111111RRRRR|wwwww10011100110
+:fnmfs.s R0004, R1115, R2731 is R0004 & op0510=0x3F & R1115; R2731 & op1626=0b10011100110
+{
+	R2731 = -1 f* ((R1115 f* R0004) f- R2731);
+}
+
 # MADDF.S reg1, reg2, reg3, reg4 - rrrrr111111RRRRR|wwwww101W00WWWW0
 :maddf.s R0004, R1115, R2731, reg4 is R0004 & op0510=0x3F & R1115; R2731 & op2426=0b101 & op2122=0b00 & op1616=0 & reg4
 {
-- 
2.39.0

